### PR TITLE
Added Good Test data for use with unit testing.

### PR DESCRIPTION
**Pull request by PompeiaPaetenari**
It was observed during another PR that there are no happy-path tests run for any of the configuration modules.  I have added a folder structure with data that is good to pass all our current modules.

Additionally, I renamed all the test functions to match the `test_what_result()` method we use in-house.

Currently, I set the `testData()` function to default to `.bad` as a testType.  If it would be more clear/preferred to not have the default parameter and just supply `.bad` in those cases.  I am fine with making that change.

### CODE REVIEW

**Comment by PompeiaPaetenari** [comment]
Just for reference of how the Good structure looks:

```
TestData/
├── Bad
│   ├── BadUnitTests
│   │   └── DanglingFile
│   │       └── BadUnitTests.swift
│   ├── ItemsOutOfOrder
│   │   ├── First.swift
│   │   └── Second.swift
│   ├── MisplacdFile
│   │   └── ThisFileIsMisplaced.swift
│   └── main.swift
├── Bad.xcodeproj
│   ├── project.pbxproj
│   └── project.xcworkspace
│       ├── contents.xcworkspacedata
│       └── xcshareddata
│           └── IDEWorkspaceChecks.plist
├── Good
│   ├── AppDelegate.swift
│   ├── Assets.xcassets
│   │   ├── AppIcon.appiconset
│   │   │   └── Contents.json
│   │   └── Contents.json
│   ├── Base.lproj
│   │   ├── LaunchScreen.storyboard
│   │   └── Main.storyboard
│   ├── Info.plist
│   ├── ViewController.swift
│   ├── main.swift
│   └── xcconfigs
│       ├── Common.xcconfig
│       ├── Debug.xcconfig
│       └── Release.xcconfig
└── Good.xcodeproj
    ├── project.pbxproj
    └── project.xcworkspace
        ├── contents.xcworkspacedata
        └── xcshareddata
            └── IDEWorkspaceChecks.plist
```

File: Tests/xcprojectlint-packageTests/NoDanglingFilesTests.swift [hunk 1]
```diff
@@ -16,9 +16,22 @@
...
+      let project = try Project(testData, errorReporter: errorReporter)
+
+      XCTAssertEqual(checkForDanglingSourceFiles(project, errorReporter: errorReporter), EX_OK)
+    } catch {
+      print(error.localizedDescription)
+      XCTFail("Failed to initialise test")
+    }
+  }
+
+  func test_danglingSourceFiles_returnError() {
```
Comment (erkekin) [inline Tests/xcprojectlint-packageTests/NoDanglingFilesTests.swift]: return**s**Error

**Comment by erkekin** [comment]
Could u remove `Info.plist` from the Good project? It is not needed as not present in the Bad one.

**Comment by PompeiaPaetenari** [comment]
> Could u remove `Info.plist` from the Good project? It is not needed as not present in the Bad one.

The last time I did that, Xcode had a hernia and refused to build.  Are we not needing to worry if Xcode can/not build it and just have a representative file structure?

**Pull request merged**